prompt/prompt-executor/prompt-executor-clients/prompt-executor-bedrock-client/src/jvmMain/kotlin/ai/koog/prompt/executor/clients/bedrock/BedrockLLMClient.jvm.kt

```diff
@@ -0,0 +1,645 @@
+            model.id.startsWith("meta.llama") -> createLlamaRequest(prompt, model)
+            else -> throw IllegalArgumentException("Model ${model.id} does not support streaming")
+        }
+
+        val streamRequest = InvokeModelWithResponseStreamRequest {
+            this.modelId = model.id
+            this.contentType = "application/json"
+            this.accept = "*/*"
+            this.body = requestBody.toString().encodeToByteArray()
+        }
+        logger.debug { "Bedrock InvokeModelWithResponseStream Request (JVM): ModelID: ${model.id}, Body: $requestBody" }
+
+        return channelFlow {
+            try {
+                bedrockClient.invokeModelWithResponseStream(streamRequest) { response: InvokeModelWithResponseStreamResponse ->
+                    response.body?.collect { event: ResponseStream ->
+                        val chunkBytes = event.asChunk()?.bytes
+                        if (chunkBytes != null) {
+                            val chunkJsonString = chunkBytes.decodeToString()
+                            send(chunkJsonString)
+                            logger.trace { "Bedrock Stream Chunk for model ${model.id} (JVM): $chunkJsonString" }
+                        } else {
+                            logger.warn { "Received null chunk bytes in stream for model ${model.id} (JVM)" }
+                        }
+                    }
+                }
+            } catch (e: Exception) {
+                logger.error(e) { "Error in Bedrock streaming for model ${model.id} (JVM)" }
+                close(e)
+            }
+        }.map { chunkJsonString ->
+            try {
+                if (chunkJsonString.isBlank()) return@map ""
+
+                when {
+                    model.id.startsWith("anthropic.claude") -> parseAnthropicStreamChunk(chunkJsonString)
+                    model.id.startsWith("amazon.titan") -> parseTitanStreamChunk(chunkJsonString)
+                    model.id.startsWith("meta.llama") -> parseLlamaStreamChunk(chunkJsonString)
+                    else -> ""
+                }
+            } catch (e: Exception) {
+                logger.warn(e) { "Failed to parse Bedrock stream chunk (JVM): $chunkJsonString" }
+                ""
+            }
+        }
+    }
+
+    // Anthropic Claude specific methods
+    private fun createAnthropicRequest(prompt: Prompt, model: LLModel, tools: List<ToolDescriptor>): JsonObject {
+        val messages = mutableListOf<JsonObject>()
+        val systemMessages = mutableListOf<String>()
+
+        prompt.messages.forEach { msg ->
+            when (msg) {
+                is Message.System -> systemMessages.add(msg.content)
+                is Message.User -> {
+                    // Check for image content if present
+                    if (msg.mediaContent.isNotEmpty()) {
+                        require(model.capabilities.contains(LLMCapability.Vision.Image)) {
```

Comment:
@kaeawc could you please clarify this check for me? As far as I understand, the model can return some media type, that is not an image, but we explicitly check for Image capabilities here. Please correct me if I miss anything.